experimental/waveshare_eink/linux_kernel_modules/fb_waveshare_eink.c

```diff
@@ -359,6 +361,12 @@ static int ws_eink_update_display(struct ws_eink_fb_par *par)
 		return ret;
 
 	memcpy(&ssbuf, &vmem, sizeof(vmem));
+	
```

Comment:
This is wrong. `ssbuf`, `vmem` and `par->info->screen_base` are all just pointers to the same chunk of memory.  What we need to do is make a copy of the data in `par->info->screen_base` and then call `bitrev8` on every byte of it.  What this code is doing is calling `bitrev8` on the original data.  I suggest allocating another block of memory at the time that vmem is allocated in probe.  Then you can store a pointer to the memory in `struct ws_eink_fb_par`